### DISCUSSION TITLE
Power Balancing Changes for 0.4.0

### DISCUSSION THREAD

**Discussion by merlerm** [NillerMedDild/Enigmatica6#1213] [Ideas]
Suggestions about nerfs/buffs go here so that we can look into it.
Mekanism nerfs are already implemented in https://github.com/NillerMedDild/Enigmatica6/pull/1125

-Buff IE refineries?
-Nerf Powah generators (besides reactor)?

**Comment by MuteTiefling** [comment]
Do note, mekanism nerfs are not set in stone and, in fact, are likely to change a bit when some new controls come to Mek's configs.

Also, bringing this over from the other issue tracker so people don't need to re-test things. these numbers are current in 0.3.x

---------------------------------------

Just to put a few numbers out there, here are some of the bigger generators in the pack. 

| Energy Producer                       | Fuel Input                                 | Energy Output (MFE) | Rate (kFE/t) | Rate (FE/t) |
|---------------------------------------|--------------------------------------------|---------------------|--------------|-------------|
| Basic Fission + Basic Turbine (quest) | 8000 mB Fissile Fuel (8 uranium ingots)    | 228.28              | 570.91       | 570910      |
| Powah Reactor (Starter)               | 8 Uraninite (plus coal, redstone, dry ice) | 26.47               | 1.56         | 1560        |
| Powah Reactor (Basic)                 | 8 Uraninite (plus coal, redstone, dry ice) | 84.75               | 6.3          | 6300        |
| Powah Reactor (Hardened)              | 8 Uraninite (plus coal, redstone, dry ice) | 148.34              | 13           | 13000       |
| Powah Reactor (Blazing)               | 8 Uraninite (plus coal, redstone, dry ice) | 181.66              | 18           | 18000       |
| Powah Reactor (Niotic)                | 8 Uraninite (plus coal, redstone, dry ice) | 357.68              | 42           | 42000       |
| Powah Reactor (Spirited)              | 8 Uraninite (plus coal, redstone, dry ice) | 447.49              | 59           | 59000       |
| Powah Reactor (Nitro)                 | 8 Uraninite (plus coal, redstone, dry ice) | 572.35              | 84           | 84000       |
| Gas Burning Generator                 | 1 bucket of Ethylene                       | 11.26               | 72.19        | 72190       |
| Mycelial Reactor                      | All the things                             |                     | 25000        | 25000000    |
| Solar Panel (Starter)                 |                                            |                     | 0.005        | 5           |
| Solar Panel (Basic)                   |                                            |                     | 0.012        | 12          |
| Solar Panel (Hardened)                |                                            |                     | 0.04         | 40          |
| Solar Panel (Blazing)                 |                                            |                     | 0.09         | 90          |
| Solar Panel (Niotic)                  |                                            |                     | 0.22         | 220         |
| Solar Panel (Spirited)                |                                            |                     | 0.5          | 500         |
| Solar Panel (nitro)                   |                                            |                     | 1.2          | 1200        |
| Thermo Generator (Starter)            | water                                      |                     | 0.008        | 8           |
| Thermo Generator (Basic)              | water                                      |                     | 0.56         | 560         |
| Thermo Generator (Hardened)           | water                                      |                     | 1.44         | 1440        |
| Thermo Generator (Blazing)            | water                                      |                     | 3.2          | 3200        |
| Thermo Generator (Niotic)             | water                                      |                     | 7.2          | 7200        |
| Thermo Generator (Spirited)           | water                                      |                     | 16           | 16000       |
| Thermo Generator (Nitro)              | water                                      |                     | 38.4         | 38400       |
| Furnator (Nitro)                      | burnable fuel                              |                     | 4.8          | 4800        |
| Wind Generator                        |                                            |                     | 0.192        | 192         |
| Solar Generator                       |                                            |                     | 0.0218       | 21.8        |
| Advanced Solar Generator              |                                            |                     | 0.1308       | 130.8       |

I've left out Thermal entirely because it's just a straight 640 rf/t at best. the only thing that changes is efficiency. 

My first, and probably biggest issue with balancing of Powah against other things is the cost. Tiered crafts suck. 

I know, I know, free power is often made expensive in MC because it's boring and spamming it is bad. But it gets spammed anyway. But just compare the cost of a niotic solar panel to a mekanism Advanced generator. There's a lot more going into that Powah panel and just not a lot of gains. 

So I do feel the recipes on these should be reworked such that they're no longer tiered upgrades, similar to the reactors. I think the range of power could also be made narrower, with the low tier generating maybe around 100 rf/t and top tier finishing at 1k rf/t

On the topic of free power... Thermo Generators. Neat concept, but it's just an RTG that needs a slow trickle of water. I don't think these should be as good as they are. Yes, they're capped on output, so even though the nitro _can_ generate 38k/t, you can only extract 10k/t. That's still kinda silly compared to solars that run for half the day and only output 1.2k/t at the same tier. 


Next, the elephant in the room. Ethylene. Obviously we can nerf it into the ground. The GBG can not currently have it's rate reduced, however, so it can do 72k no matter how much we nerf ethylene. Reducing the amount of bio-fuel you get per crop could be a good start here, as well as increasing the cost of production. If you only get 5k net rf/t (or less) out of the production line, then it suddenly becomes a lot less interesting as permanent power. It instead becomes a nice way to buffer a backup, or as a portable energy source before getting to wireless stuff. A GBG and a tank of gas is a great way to power a digital miner before you can get entangloporters or Powah wireless stuff going. 

So, I'd say we keep energy density where its at. 1 bucket making 11.26 million RF. But try to tweak it so the production is much more costly. By comparison, LPG in a compression dynamo currently outputs a total of 18 million RF. But there's a fair bit more work that goes into producing that and, again, it's capped at 640rf/t. 

Powah Reactors seem... lackluster. You end up going through a lot of automation to actually supply coal/redstone/ice/uranium to keep them running. Is it on the same order as a mekanism fissile fuel? Probably not. But given work that is involved, and the relative rarity of uranium, it feels like they need a boost. 

I'm thinking a boost to energy density would be great for them.  Just let it run longer so you're not out hunting for uranium as often at early stages. I think narrowing the gap between tiers could be good here too. Maybe 5k-80k rather than the current 1.5k-72k. 

Finally, Mek Fission/Fusion.

The gadget patterns I set up for the quests for Fission will handle a burn rate of 20mb/t with water cooling, and that turbine will output 570k fe/t. That RF/t is strictly determined by the size of the turbine and the steam produced. A bigger turbine, while a little costly, isn't that difficult to put together. Neither is a bigger reactor to get better steam production. 

The main issue with fission is the waste management, which ultimately means running it constantly isn't possible. There's just no way to remove the waste fast enough, so you end up running out of room to store it. It's currently just a speed-bump/complicated crafting step to get to Fusion and the Mekasuit. Fusion just gets silly, as we all know. And in this pack there's just nowhere to spend it all. Sure, you can dump it into an SPS, but you're mostly done with that by the time you can even make a fusion reactor since there's not much use for antimatter. Having some custom replicator recipes for antimatter could be interesting, however. Even for a non-expert mode, as it could give people a legitimate reason to actually create a full processing chain and a dump for the power they generate. The question, then, is what do we add as recipes for nucleosynthesis? It currently costs 10 uranium ingots to make 1 mb of antimatter, so they would need to be really good to be interesting. 

Since Fission and Fusion are both ultimately tied to the turbine, I think that's the place to look to bringing them in line. I don't know if there's a simple dial we can tweak to reduce the RF:steam ratio. It seems like it would be a mess to try to balance otherwise since fusion can just produce such an ungodly amount of steam. 


Now... the thing that I think is probably most important here is that we have basically nothing that acts as a constant heavy drain on power. There's little upkeep cost on a base. Yes some machines can be expensive to run, but you run them so infrequently that it doesn't amount to much in terms of RF/t over their lifetime. That's why I'm thinking some nucleosynthesis recipes could be nice since they can really drain a ton of power. 

Otherwise, RS is the only thing I can even think of that actually has a constant upkeep cost. And that is pretty tame. I don't think making it more costly is the way to go, however, as that just makes it inaccessible early on.

**Reply by merlerm** [reply]
By the way, this table looks really cool and is very useful imo. Could go on the wiki on the main power changes page?

**Reply by MuteTiefling** [reply]
Sure thing. I'll have to spend some time updating it, as this represented our baseline from before my changes.
